sematic/resolvers/local_resolver.py

```diff
@@ -139,14 +139,18 @@ def _future_will_schedule(self, future: AbstractFuture) -> None:
         super()._future_will_schedule(future)
 
         run = self._populate_run_and_artifacts(future)
-
-        run.future_state = FutureState.SCHEDULED
+        self._update_run_and_future_pre_scheduling(run, future)
         run.root_id = self._futures[0].id
-        run.started_at = datetime.datetime.utcnow()
 
         self._add_run(run)
         self._save_graph()
 
+    def _update_run_and_future_pre_scheduling(self, run: Run, future: AbstractFuture):
+        """Perform any updates to run before saving it to the DB pre-scheduling"""
+        future.state = FutureState.SCHEDULED
+        run.future_state = FutureState.SCHEDULED
+        run.started_at = datetime.datetime.utcnow()
+
```

Comment:
I wonder if it's cleaner to simply have two different implementations of `_future_will_schedule` on `LocalResolver` and `CloudResolver, instead of this type of branching.

Reply:
I think there's enough overlap to leave it as-is